apps/wpcom-block-editor/src/calypso/iframe-bridge-server.js

```diff
@@ -17,6 +17,50 @@ import tinymce from 'tinymce/tinymce';
  */
 import { inIframe, sendMessage } from './utils';
 
+/**
+ * Monitors Gutenberg for when an editor is opened with content originally authored in the classic editor.
+ *
+ * @param {MessagePort} calypsoPort Port used for communication with parent frame.
+ */
+function triggerConversionPrompt( calypsoPort ) {
+	const { port1, port2 } = new MessageChannel();
+
+	const unsubscribe = subscribe( () => {
+		const currentPost = select( 'core/editor' ).getCurrentPost();
+		const initialized = currentPost && currentPost.id;
+
+		if ( ! initialized ) {
+			return;
+		}
+
+		unsubscribe();
+
+		const blocks = select( 'core/editor' ).getBlocks();
+		const eligible = blocks.length === 1 && blocks[ 0 ].name === 'core/freeform';
+
+		if ( ! eligible ) {
+			return;
+		}
+
+		calypsoPort.postMessage( { action: 'triggerConversionRequest' }, [ port2 ] );
+
+		port1.onmessage = ( { data: confirmed } ) => {
+			port1.close();
+
+			if ( confirmed !== true ) {
+				return;
+			}
+
+			dispatch( 'core/editor' ).replaceBlock(
+				blocks[ 0 ].clientId,
+				rawHandler( {
+					HTML: blocks[ 0 ].originalContent,
+				} )
+			);
+		};
+	} );
```

Comment:
🤟

Reply:
it's beautiful 😭thank you for all your help